astropy/io/ascii/ui.py

```diff
@@ -220,7 +220,7 @@ def _guess(table, read_kwargs, format, fast_reader):
             reader.guessing = True
             return reader.read(table)
 
-        except (core.InconsistentTableError, ValueError, TypeError,
+        except (core.InconsistentTableError, ValueError, TypeError, AttributeError,
```

Comment:
@embray - There is already an `InconsistentTableError` for exactly that purpose, and in fact I went so far as to start writing a new issue saying that, in general, `io.ascii` readers should catch any errors internally and re-raise that.  But then I wasn't sure and decided to let sleeping dogs lie (which as the owner of vigorous puppy is extremely important).  This would be a big-ish fix and maybe it would be a good thing, but it would definitely involve a lot of try/except in the reader code.  I actually thing the biggest benefit would be in being able to re-raise more useful errors for uses.

In any case I think that given the precedent of catching some errors outside of `InconsistentTableError`, that this particular patch is OK for now.